token/services/sql/init.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package sql
+
+import (
+	"crypto/sha256"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"regexp"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/flogging"
+	"github.com/pkg/errors"
+)
+
+var logger = flogging.MustGetLogger("token-sdk.sql")
+
+var Transactions *TransactionDB
+var Tokens *TokenDB
+
+func Init(driverName, dataSourceName, tablePrefix, name string, createSchema bool, maxOpenConns int) error {
+	logger.Infof("connecting to sql database [%s:%s]", driverName, tablePrefix) // dataSource can contain a password
+	if Transactions != nil || Tokens != nil {
+		// return errors.New("database can only be initialized once") // TODO: how do we handle this?
```

Comment:
This one is about having several TMS and database instances. We generate a suffix for the table names based on the 'name' (i.e. we're hashing the TMS id), see getTableNames in this same file. It's an ugly solution and I hope we can get rid of it. More so because there is only one global instance of each DB in the package, so you're not even able to instantiate a second one for another TMS.

More logical is if the user configures the prefix in the TMS config (like we discussed earlier). I'm not sure what that means for the global instances in this package, we should discuss.

Reply:
Notice that we use this only for the tests. For the rest, each TMS defines it is own databases with their locations. 
I think we should be good. What do you think?

Reply:
Right, I took your PR for a spin now, and noticed that the user can define all the settings per db, including the tableprefix. This removes the need for the suffix, so let's remove it.

I still have my doubts about this level of configurability. Why don't we always use the same database, tableprefix, maxOpenConnections for all the sql stores in a TMS? Setting it in one place will make it easier to configure and understand, and harder to make mistakes.

Reply:
Yes, indeed. I was planning to introduce a unified driver that offers instantiations for all databases backing them with a single sql db instance. Shall I add it here?